### ISSUE TITLE
Proton for sourcemods

### ISSUE THREAD

**Issue by XakepSDK** [ValveSoftware/steam-for-linux#6036]
#### Your system information

* Steam client version (build number or date): Jan 19 2019, at 02:01:59
* Distribution (e.g. Ubuntu): Arch Linux
* Opted into Steam client beta?: [Yes/No] Yes
* Have you checked for system updates?: [Yes/No] Yes

#### Please describe your issue in as much detail as possible:
Some sourcemods don't support linux. I want to play them, but steam won't launch them with proton.
For example: HL2: Aftermath or HL2: MMod

#### Steps for reproducing this issue:

1. Install sourcemod
2. Set HL2 or HL2 episodes settings to use Proton
3. Set sourcemod settings to use Proton
4. Launch game
5. Native game starts, instead of proton version.

**Comment by Matthew-Bradley** [comment]
> Isn't #5677 already fixed with latest beta?

I don't believe that applies here. We are not trying to run a program that is external to Steam.

What we need is to install the Windows version of a game instead of the Linux version, when both exist in the Steam store/library. This cannot currently be accomplished as Linux native versions are always installed when available. This remains regardless of what Proton settings are chosen in the game's settings.

By emulating the Windows version with Proton instead of running the native Linux version, we would be able to link against windows DLL files provided as part of a sourcemod. If only the native Linux version can be installed on Linux, then attempting to run a sourcemod with DLLs compiled for windows will result in an error. Most sourcemods only provide their content compiled as DLLs for Windows and few offer SOs for Linux.

It's a surprising issue when the most difficult games to run on Steam for Linux are mods for a game with a native linux version.

**Comment by stopmotio** [comment]
This would be the worst temporary fix of all, but try installing steam under wine or lutris and installing to that.

**Comment by Matthew-Bradley** [comment]
This seems like this should be a straightforward fix for a problem that should never have happened.

Just actually honor the per-game compatibility setting for steamapps in properties->compatibility. Yes, this issue is actually a bug not a feature request.

If someone ticks "Force the use of a specific steam play compatibility too" launch the windows version of HL2 under proton instead of the native linux runtime. Then the "could not load library client" error wouldn't occur and every sourcemod would work on linux instead of almost none of them.

Sourcemods compiled into *.DLLs could be configured to run correctly, and those compiled into *.so files can remain on the default setting which uses a version of HL2 compiled for linux.

If more detailed steps to reproduce the problem are needed I'd be happy to provide them.

**Comment by Matthew-Bradley** [comment]
Bumping.

Please honor the per-game compatibility setting for steamapps in properties->compatibility. Right now it has no effect. Yes, this issue is a bug not a feature request.

If someone ticks "Force the use of a specific steam play compatibility too" launch the windows version of HL2 under proton instead of the native linux runtime. Then the "could not load library client" error shouldn't occur.

At the moment almost NO sourcemods work on linux because of this issue.

**Comment by Matthew-Bradley** [comment]
If a detailed series of steps to replicate this bug using a specific sourcemode are required I would be happy to provide them

**Comment by Matthew-Bradley** [comment]
If a detailed series of steps to replicate this bug (using a specific sourcemod) are required I would be happy to provide them.

**Comment by dangerousones** [comment]
I can also replicate this issue, I'm unable to play any sourcemod, including "MMod" on Ubuntu Linux 20.04 Desktop.

**Comment by Matthew-Bradley** [comment]
@kisak-valve Is there additional information which would make this issue easier to resolve?

**Comment by Unaccounted4** [comment]
Still present 01/08/23.

**Comment by facespkz** [comment]
Currently able to launch MMod without any issues. In my experience, sourcemods generally inherit the runtime/Proton of the app/game that they're based off of. Most of them run on Source SDK Base 2013, so that needs to be set to use Proton as well.

If the app/game is already installed, there may be an update available. That would be the platform-specific binaries.

**Comment by owlshrimp** [comment]
> Currently able to launch MMod without any issues. In my experience, sourcemods generally inherit the runtime/Proton of the app/game that they're based off of. Most of them run on Source SDK Base 2013, so that needs to be set to use Proton as well.
> 
> If the app/game is already installed, there may be an update available. That would be the platform-specific binaries.

This was possibly(?) fixed a year or so ago. previously sourcemods needed to be compiled to .so instead of .dll

It also depends on the sourcemod. if the mod doesn't include any compiled code and only has models/textures/sounds, then this is a non-issue.
